pkg/mimir/modules.go

```diff
@@ -970,7 +970,12 @@ func (t *Mimir) initAlertManager() (serv services.Service, err error) {
 	}
 	features, err := featurecontrol.NewFlags(util_log.Logger, mode)
 	util_log.CheckFatal("initializing Alertmanager feature flags", err)
-	compat.InitFromFlags(util_log.Logger, features)
+
+	compatLogger := log.NewNopLogger()
+	if t.Cfg.Alertmanager.LogParsingLabelMatchers {
+		compatLogger = util_log.Logger
+	}
+	compat.InitFromFlags(compatLogger, features)
```

Comment:
This is where the logger is initialized for the label matchers parser code. You can find it [here](https://github.com/grafana/mimir/blob/mimir-2.13.0/vendor/github.com/prometheus/alertmanager/matchers/compat/parse.go#L59).